### DISCUSSION TITLE
Error when using example in documentation

### DISCUSSION THREAD

**Discussion by davidlag0** [panva/jose#441] [Q&A] [answered]
Hi,

First, thank you for this library! I'm really glad someone put in the time to build all these functionalities into a simple library to use!

I'm trying to use this example in the documentation that perfectly fits my use case:
https://github.com/panva/jose/blob/main/docs/classes/jwt_encrypt.EncryptJWT.md

`secretKey` is defined as such in my code:
`
  const secretKey = await hkdf(
    "sha256",
    secret,
    "",
    "NextAuth.js Generated Encryption Key",
    32
  );
`

When I use `console.log` to show this value, I get this output in the console:
`  console.log
    secretKey: Uint8Array(32) [
      203, 75, 118, 123, 149, 238, 125, 187,
      130, 10, 134,  78,  98, 203, 124, 109,
      170, 48, 106, 210, 160, 135,  23, 165,
        5, 40, 102,  85, 202, 202,  94, 149
    ]`

And when the example code is run, I get this error message:
```
    TypeError: plaintext must be an instance of Uint8Array

      29 |     .setAudience("urn:example:audience")
      30 |     .setExpirationTime("2h")
    > 31 |     .encrypt(encryptionSecret);
         |      ^
      32 |
      33 |   console.log("jwt", jwt);
      34 |

      at new FlattenedEncrypt (node_modules/jose/dist/node/cjs/jwe/flattened/encrypt.js:17:19)
      at new CompactEncrypt (node_modules/jose/dist/node/cjs/jwe/compact/encrypt.js:7:27)
      at EncryptJWT.encrypt (node_modules/jose/dist/node/cjs/jwt/encrypt.js:49:21)
      at encrypt (lib/session.js:31:6)
```

I am running the very same code there is in the documentation I provided the link for above:
```
  const jwt = await new jose.EncryptJWT({ "urn:example:claim": true })
    .setProtectedHeader({ alg: "dir", enc: "A256GCM" })
    .setIssuedAt()
    .setIssuer("urn:example:issuer")
    .setAudience("urn:example:audience")
    .setExpirationTime("2h")
    .encrypt(encryptionSecret);

  console.log("jwt", jwt);
```

I am currently using the last version available from npm: 4.9.0.

How could I resolve this issue?

Thank you very much!

**Comment by panva** [comment]
Are you running this directly in Node.js? In case you don't, run this issue by whomever orchestrates your node runtime. This is a common issue nowadays when people run their code in a vm (node vm builtin) but don't correctly pass global javascript constructors, which in tirn makes the instanceof operator fail like it does for you.

**Reply by davidlag0** [reply]
I'm running this through Jest as I'm looking to encrypt a JWT for NextAuth.js to simulate authenticated tests.

You made me realize I have this in my `jest.config.js`:
```
  testEnvironment: "jest-environment-jsdom",
  moduleNameMapper: {
    "^jose$": require.resolve("jose"),
    "^@panva/hkdf$": require.resolve("@panva/hkdf"),
    "^uuid$": require.resolve("uuid"),
  },
```

If I comment the two lines for "jose" and "@panva/hkdf", I then get this error message from Jest:
```
    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /<redacted>/node_modules/@panva/hkdf/dist/web/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import derive from './runtime/hkdf.js';
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

       8 |   return hkdf("sha256", secret, "", "NextAuth.js Generated Encryption Key", 32);
       9 | }
    > 10 |
         | ^
      11 | // Function logic derived from https://github.com/nextauthjs/next-auth/blob/5c1826a8d1f8d8c2d26959d12375704b0a693bfc/packages/next-auth/src/jwt/index.ts#L16-L25
      12 | export async function encode(claims, secret) {
      13 |   const maxAge = 30 * 24 * 60 * 60; // 30 days

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1796:14)
      at Object.<anonymous> (lib/session.js:10:52)
```

Considering this now, I will check other discussions to see if I can find anything to help me.

**Comment by davidlag0** [comment, marked as answer]
@panva 
After additional research following your pointers, I did some more tests such as changing the overall environment. It lead to different issues with other tests I have for the frontend part of the app so I couldn't do this. I ended up keeping the very same Jest config because I came across this: https://stephenweiss.dev/jest-test-environments

This lead me to add this at the top of the file with my tests that require a node environment:
```
/**
 * @jest-environment node
 */
```

Thank you very much for your quick answer and pointers, it helped a lot to help me figure out a solution!
